sdk/servicebus/azure-servicebus/azure/servicebus/management/_models.py

```diff
@@ -521,83 +693,138 @@ def _to_internal_entity(self):
-        subscription.message_count = internal_subscription.message_count
-        subscription.created_at = internal_subscription.created_at
-        subscription.updated_at = internal_subscription.updated_at
-        subscription.accessed_at = internal_subscription.accessed_at
-        subscription.message_count_details = internal_subscription.message_count_details
 
         return subscription
 
+    @property
+    def name(self):
+        """Name of subscription
+
+        :rtype: str
+        """
+        return self._name
+
+    @property
+    def accessed_at(self):
+        """Last time a message was sent, or the last time there was a receive request
+
+        :rtype: ~datetime.datetime
+        """
+        return self._internal_sd.accessed_at
+
+    @property
+    def created_at(self):
+        """The exact time the subscription was created.
+
+        :rtype: ~datetime.datetime
+        """
+        return self._internal_sd.created_at
+
+    @property
+    def updated_at(self):
+        """The exact time the entity is updated.
+
+        :rtype: ~datetime.datetime
+        """
+        return self._internal_sd.updated_at
+
+    @property
+    def total_message_count(self):
+        """The number of messages in the subscription.
+
+        :rtype: int
+        """
+        return self._internal_sd.message_count
+
+    @property
+    def active_message_count(self):
+        """Number of active messages in the subscription.
+
+        :rtype: int
+        """
+        return self._internal_sd.message_count_details.active_message_count
 
-class RuleDescription(object):
-    """Description of a topic subscription rule.
+    @property
+    def dead_letter_message_count(self):
+        """Number of messages that are dead lettered.
+
+        :rtype: int
+        """
+        return self._internal_sd.message_count_details.dead_letter_message_count
+
+    @property
+    def transfer_dead_letter_message_count(self):
+        """Number of messages transferred into dead letters.
+
+        :rtype: int
+        """
+        return self._internal_sd.message_count_details.transfer_dead_letter_message_count
+
+    @property
+    def transfer_message_count(self):
+        """Number of messages transferred to another queue, topic, or subscription.
+
+        :rtype: int
+        """
+        return self._internal_sd.message_count_details.transfer_message_count
+
+
+class RuleProperties(DictMixin):
+    """Properties of a topic subscription rule.
 
     :param name: Name of the rule.
     :type name: str
-    :keyword filter: The filter of the rule.
-    :type filter: Union[~azure.servicebus.management.models.CorrelationRuleFilter,
-     ~azure.servicebus.management.models.SqlRuleFilter]
-    :keyword action: The action of the rule.
-    :type action: Optional[~azure.servicebus.management.models.SqlRuleAction]
-    :keyword created_at: The exact time the rule was created.
+    :ivar filter: The filter of the rule.
```

Comment:
ivar and not keyword?  I don't actually know what our policy for this is, so feel free to lmk if you're aware/this is intentional.

Reply:
"filter" and many other attributes are both ivar and keyword. I changed to ivar because we don't expect a user to create a RuleProperties instance using the keywords, even though they can if they really do so.